src/client/ua_client_highlevel_subscriptions.c

```diff
@@ -192,6 +192,7 @@ UA_Client_Subscriptions_addMonitoredEvent(UA_Client *client, const UA_UInt32 sub
     UA_LOG_DEBUG(client->config.logger, UA_LOGCATEGORY_CLIENT,
                  "Created a monitored item with client handle %u", client->monitoredItemHandles);
 
+//    UA_EventFilter_delete(evFilter);
```

Comment:
@Pro do you have an idea how to handle this correctly?

Without that line valgrind complains about:
`
^C[08/18/2017 09:45:53.260] info/userland       received ctrl-c
[08/18/2017 09:45:54.034] info/userland Subscription removed
==32489== 
==32489== HEAP SUMMARY:
==32489==     in use at exit: 208 bytes in 4 blocks
==32489==   total heap usage: 354 allocs, 350 frees, 666,233 bytes allocated
==32489== 
==32489== 208 (32 direct, 176 indirect) bytes in 1 blocks are definitely lost in loss record 4 of 4
==32489==    at 0x4C2FB55: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32489==    by 0x403F2F: UA_new (ua_types.c:821)
==32489==    by 0x407F6A: UA_EventFilter_new (ua_types_generated_handling.h:4491)
==32489==    by 0x408B87: UA_Client_Subscriptions_addMonitoredEvent (ua_client_highlevel_subscriptions.c:141)
==32489==    by 0x401ACD: main (tutorial_client_events.c:109)
==32489== 
==32489== LEAK SUMMARY:
==32489==    definitely lost: 32 bytes in 1 blocks
==32489==    indirectly lost: 176 bytes in 3 blocks
==32489==      possibly lost: 0 bytes in 0 blocks
==32489==    still reachable: 0 bytes in 0 blocks
==32489==         suppressed: 0 bytes in 0 blocks
==32489== 
==32489== For counts of detected and suppressed errors, rerun with: -v
==32489== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
`

Which makes sense. So i added the "UA_EventFilter_delete(evFilter);". Now valgrind complains about:
`
[08/18/2017 09:45:06.598] info/userland Create subscription succeeded, id 394
==32114== Invalid free() / delete / delete[] / realloc()
==32114==    at 0x4C2EDEB: free (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32114==    by 0x401EF6: String_deleteMembers (ua_types.c:92)
==32114==    by 0x4043DE: deleteMembers_noInit (ua_types.c:977)
==32114==    by 0x40447A: UA_deleteMembers (ua_types.c:991)
==32114==    by 0x4046D5: UA_Array_delete (ua_types.c:1058)
==32114==    by 0x404428: deleteMembers_noInit (ua_types.c:983)
==32114==    by 0x40447A: UA_deleteMembers (ua_types.c:991)
==32114==    by 0x4046D5: UA_Array_delete (ua_types.c:1058)
==32114==    by 0x404428: deleteMembers_noInit (ua_types.c:983)
==32114==    by 0x40447A: UA_deleteMembers (ua_types.c:991)
==32114==    by 0x407F92: UA_EventFilter_deleteMembers (ua_types_generated_handling.h:4501)
==32114==    by 0x408E99: UA_Client_Subscriptions_addMonitoredEvent (ua_client_highlevel_subscriptions.c:195)
==32114==  Address 0x438998 is in a r-x mapped file /home/fmee/projects/foss/build-open62541-Desktop-Default/bin/examples/tutorial_client_events segment
==32114==
`